cli/src/main.rs

```diff
@@ -1,11 +1,20 @@
 use clap::Parser;
 use schema2000::{render_schema, SchemaHypothesis};
-
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let _args = Args::parse();
-
-    let stdin = std::io::stdin();
-    let stdin = stdin.lock();
+use std::error::Error;
+use std::fs::File;
+use std::io::{self, Read};
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let args = Args::parse();
+
+    let stdin: Box<dyn Read> = if let Some(file_path) = args.file {
```

Comment:
Maybe we should name this `reader` or something? Because it is either stdin or a file.

Reply:
Done 👍